### ISSUE TITLE
Not able to invoke  ScuBA successfully. Getting errors

### ISSUE THREAD

**Issue by ramarrgit** [cisagov/ScubaGear#901]
Did the following in the same order"
---------------------------------------------------

Set-ExecutionPolicy RemoteSigned
.\SetUp.ps1
.\opa_windows_amd64.exe version
Import-Module -Name .\PowerShell\ScubaGear
Invoke-SCuBA -ProductNames aad -LogIn $true -Verbose

It didn't work. No Login screen was launched.
Please help to get this audit up and running.

================================================================================

Set up messages:
----------------------------

Setting PSGallery repository to trusted.
                                                                                                                                                     
DEBUG: PowerShellGet: 2.2.5 already has latest installed.
Installed the latest acceptable version of MicrosoftTeams: 5.9.0.
ExchangeOnlineManagement: 3.2.0 updated to version 3.4.0.
Installed the latest acceptable version of Microsoft.Online.SharePoint.PowerShell: 16.0.24510.12000.
Installed the latest acceptable version of PnP.PowerShell: 1.12.0.
Installed the latest acceptable version of Microsoft.PowerApps.Administration.PowerShell: 2.0.180.
Installed the latest acceptable version of Microsoft.PowerApps.PowerShell: 1.0.34.
Microsoft.Graph.Authentication: 1.27.0 updated to version 2.13.0.
Installed the latest acceptable version of Microsoft.Graph.Beta.Users: 2.13.0.                                                                                                               
Installed the latest acceptable version of Microsoft.Graph.Beta.Groups: 2.13.0.                                                                                                              
Installed the latest acceptable version of Microsoft.Graph.Beta.Identity.DirectoryManagement: 2.13.0.                                                                                        
Installed the latest acceptable version of Microsoft.Graph.Beta.Identity.Governance: 2.13.0.                                                                                                 
Installed the latest acceptable version of Microsoft.Graph.Beta.Identity.SignIns: 2.13.0.                                                                                                    
Installed the latest acceptable version of powershell-yaml: 0.4.7.           
DEBUG: opa_windows_amd64.exe: 0.59.0 already has latest installed.                                      

================================================================================
Invoke-SCuBA messages:
-------------------------------

VERBOSE: Removing the imported "Select-DohServer" function.
VERBOSE: Removing the imported "Invoke-RobustDnsTxt" function.
VERBOSE: Removing the imported "Get-ScubaSpfRecord" function.
VERBOSE: Removing the imported "Get-ScubaDmarcRecord" function.
VERBOSE: Removing the imported "Export-EXOProvider" function.
VERBOSE: Removing the imported "Get-ScubaDkimRecord" function.
VERBOSE: Removing the imported "Get-EXOTenantDetail" function.
VERBOSE: Removing the imported "Get-AADTenantDetail" function.
VERBOSE: Removing the imported "Export-AADProvider" function.
VERBOSE: Removing the imported "Get-PrivilegedUser" function.
VERBOSE: Removing the imported "Get-PrivilegedRole" function.
VERBOSE: Removing the imported "Export-DefenderProvider" function.
VERBOSE: Removing the imported "Get-TeamsTenantDetail" function.
VERBOSE: Removing the imported "Export-TeamsProvider" function.
VERBOSE: Removing the imported "Export-SharePointProvider" function.
VERBOSE: Removing the imported "Invoke-Rego" function.
VERBOSE: Removing the imported "New-Report" function.
VERBOSE: Removing the imported "Import-SecureBaseline" function.
VERBOSE: Removing the imported "Disconnect-SCuBATenant" function.
VERBOSE: Removing the imported "Connect-Tenant" function.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Providers\ExportAADProvider.psm1'.
VERBOSE: Exporting function 'Export-AADProvider'.
VERBOSE: Exporting function 'Get-AADTenantDetail'.
VERBOSE: Exporting function 'Get-PrivilegedUser'.
VERBOSE: Exporting function 'Get-PrivilegedRole'.
VERBOSE: Importing function 'Export-AADProvider'.                                                                                                                                            VERBOSE: Importing function 'Get-AADTenantDetail'.                                                                                                                                           VERBOSE: Importing function 'Get-PrivilegedRole'.                                                                                                                                            VERBOSE: Importing function 'Get-PrivilegedUser'.                                                                                                                                            VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Providers\ExportDefenderProvider.psm1'.                             VERBOSE: Exporting function 'Export-DefenderProvider'.                                                                                                                                       VERBOSE: Importing function 'Export-DefenderProvider'.                                                                                                                                       VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Providers\ExportEXOProvider.psm1'.                                  VERBOSE: Exporting function 'Export-EXOProvider'.                                                                                                                                            VERBOSE: Exporting function 'Get-EXOTenantDetail'.                                                                                                                                           VERBOSE: Exporting function 'Select-DohServer'.                                                                                                                                              VERBOSE: Exporting function 'Invoke-RobustDnsTxt'.                                                                                                                                           VERBOSE: Exporting function 'Get-ScubaSpfRecord'.                                                                                                                                            VERBOSE: Exporting function 'Get-ScubaDkimRecord'.                                                                                                                                           VERBOSE: Exporting function 'Get-ScubaDmarcRecord'.                                                                                                                                          VERBOSE: Importing function 'Export-EXOProvider'.                                                                                                                                            VERBOSE: Importing function 'Get-EXOTenantDetail'.
VERBOSE: Importing function 'Get-ScubaDkimRecord'.
VERBOSE: Importing function 'Get-ScubaDmarcRecord'.
VERBOSE: Importing function 'Get-ScubaSpfRecord'.
VERBOSE: Importing function 'Invoke-RobustDnsTxt'.
VERBOSE: Importing function 'Select-DohServer'.
VERBOSE: Importing function 'Export-PowerPlatformProvider'.
VERBOSE: Importing function 'Get-PowerPlatformTenantDetail'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Providers\ExportSharePointProvider.psm1'.
VERBOSE: Exporting function 'Export-SharePointProvider'.
VERBOSE: Importing function 'Export-SharePointProvider'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Providers\ExportTeamsProvider.psm1'.
VERBOSE: Exporting function 'Export-TeamsProvider'.
VERBOSE: Exporting function 'Get-TeamsTenantDetail'.
VERBOSE: Importing function 'Export-TeamsProvider'.
VERBOSE: Importing function 'Get-TeamsTenantDetail'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Connection\Connection.psm1'.
VERBOSE: Importing function 'Connect-Tenant'.
VERBOSE: Importing function 'Disconnect-SCuBATenant'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\RunRego\RunRego.psm1'.
VERBOSE: Importing function 'Invoke-Rego'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\CreateReport\CreateReport.psm1'.
VERBOSE: Importing function 'Import-SecureBaseline'.
VERBOSE: Importing function 'New-Report'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\ScubaConfig\ScubaConfig.psm1'.
VERBOSE: Importing function 'Copy-ScubaBaselineDocument'.
WARNING: Error retrieving Tenant details using Get-AADTenantDetail Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaIdentityConditionalAccessPolicy. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaSubscribedSku. Object reference not set to an instance of an object.
WARNING: Omitting calls to Get-PrivilegedRole and Get-PrivilegedUser.
WARNING: Error running Get-MgBetaSubscribedSku. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaUserCount. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaPolicyAuthorizationPolicy. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaDirectorySetting. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaPolicyAdminConsentRequestPolicy. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaPolicyAuthenticationMethodPolicy. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaDomain. Object reference not set to an instance of an object.
Fatal Error involving the Report Creation.
            Ending ScubaGear execution. See the exception message for more details: Invalid JSON primitive: .
At C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Orchestrator.psm1:917 char:13
+             throw $InvokeReportErrorMessage
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Fatal Error inv...ON primitive: .:String) [], RuntimeException
    + FullyQualifiedErrorId : Fatal Error involving the Report Creation.
            Ending ScubaGear execution. See the exception message for more details: Invalid JSON primitive:

**Comment by rezabj** [comment]
I am experiencing the same issue.

**Comment by gdasher** [comment]
This error probably means that there is a null reference somewhere.  Can you try running some basic Graph commands and see if they work or provide the same error?  Say,

```
Connect-MgGraph
Get-MgBetaIdentityConditionalAccessPolicy.
```

If those work, I would suggest trying to disconnect everything and perhaps delete your graph state:

```
Disconnect-SCuBATenant
rm -r C:\Users\{username}\.graph
```

and then try again.    Would also help to see if the issue is limited to the AAD provider (graph) or extends to other providers.

**Comment by ramarrgit** [comment]
Hi,

Ran the following.

Disconnect-SCuBATenant
rm -r C:\Users\{username}\.graph

Then ran Invoke-SCuBA -ProductNames aad -LogIn $true -Verbose
I now got the logon prompt. Entered the details. But I got the errors ( I think same as before)

VERBOSE: Removing the imported "Select-DohServer" function.
VERBOSE: Removing the imported "Get-ScubaDmarcRecord" function.
VERBOSE: Removing the imported "Invoke-RobustDnsTxt" function.
VERBOSE: Removing the imported "Get-ScubaSpfRecord" function.
VERBOSE: Removing the imported "Get-ScubaDkimRecord" function.
VERBOSE: Removing the imported "Get-EXOTenantDetail" function.
VERBOSE: Removing the imported "Export-EXOProvider" function.
VERBOSE: Removing the imported "Get-PrivilegedUser" function.
VERBOSE: Removing the imported "Get-PrivilegedRole" function.
VERBOSE: Removing the imported "Get-AADTenantDetail" function.
VERBOSE: Removing the imported "Export-AADProvider" function.
VERBOSE: Removing the imported "Export-DefenderProvider" function.
VERBOSE: Removing the imported "Get-TeamsTenantDetail" function.
VERBOSE: Removing the imported "Export-TeamsProvider" function.
VERBOSE: Removing the imported "Export-SharePointProvider" function.
VERBOSE: Removing the imported "Invoke-Rego" function.
VERBOSE: Removing the imported "New-Report" function.
VERBOSE: Removing the imported "Import-SecureBaseline" function.
VERBOSE: Removing the imported "Disconnect-SCuBATenant" function.
VERBOSE: Removing the imported "Connect-Tenant" function.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Providers\ExportAADProvider.psm1'.
VERBOSE: Exporting function 'Export-AADProvider'.
VERBOSE: Exporting function 'Get-AADTenantDetail'.
VERBOSE: Exporting function 'Get-PrivilegedUser'.
VERBOSE: Exporting function 'Get-PrivilegedRole'.
VERBOSE: Importing function 'Export-AADProvider'.                                                                                                                                            VERBOSE: Importing function 'Get-AADTenantDetail'.                                                                                                                                           VERBOSE: Importing function 'Get-PrivilegedRole'.                                                                                                                                            VERBOSE: Importing function 'Get-PrivilegedUser'.                                                                                                                                            VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Providers\ExportDefenderProvider.psm1'.                             VERBOSE: Exporting function 'Export-DefenderProvider'.                                                                                                                                       VERBOSE: Importing function 'Export-DefenderProvider'.                                                                                                                                       VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Providers\ExportEXOProvider.psm1'.                                  VERBOSE: Exporting function 'Export-EXOProvider'.                                                                                                                                            VERBOSE: Exporting function 'Get-EXOTenantDetail'.                                                                                                                                           VERBOSE: Exporting function 'Select-DohServer'.                                                                                                                                              VERBOSE: Exporting function 'Invoke-RobustDnsTxt'.                                                                                                                                           VERBOSE: Exporting function 'Get-ScubaSpfRecord'.                                                                                                                                            VERBOSE: Exporting function 'Get-ScubaDkimRecord'.                                                                                                                                           VERBOSE: Exporting function 'Get-ScubaDmarcRecord'.                                                                                                                                          VERBOSE: Importing function 'Export-EXOProvider'.                                                                                                                                            VERBOSE: Importing function 'Get-EXOTenantDetail'.
VERBOSE: Importing function 'Get-ScubaDkimRecord'.
VERBOSE: Importing function 'Get-ScubaDmarcRecord'.
VERBOSE: Importing function 'Get-ScubaSpfRecord'.
VERBOSE: Importing function 'Invoke-RobustDnsTxt'.
VERBOSE: Importing function 'Select-DohServer'.
VERBOSE: Importing function 'Export-PowerPlatformProvider'.
VERBOSE: Importing function 'Get-PowerPlatformTenantDetail'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Providers\ExportSharePointProvider.psm1'.
VERBOSE: Exporting function 'Export-SharePointProvider'.
VERBOSE: Importing function 'Export-SharePointProvider'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Providers\ExportTeamsProvider.psm1'.
VERBOSE: Exporting function 'Export-TeamsProvider'.
VERBOSE: Exporting function 'Get-TeamsTenantDetail'.
VERBOSE: Importing function 'Export-TeamsProvider'.
VERBOSE: Importing function 'Get-TeamsTenantDetail'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Connection\Connection.psm1'.
VERBOSE: Importing function 'Connect-Tenant'.
VERBOSE: Importing function 'Disconnect-SCuBATenant'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\RunRego\RunRego.psm1'.
VERBOSE: Importing function 'Invoke-Rego'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\CreateReport\CreateReport.psm1'.
VERBOSE: Importing function 'Import-SecureBaseline'.
VERBOSE: Importing function 'New-Report'.
VERBOSE: Loading module from path 'C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\ScubaConfig\ScubaConfig.psm1'.
VERBOSE: Importing function 'Copy-ScubaBaselineDocument'.
WARNING: Error retrieving Tenant details using Get-AADTenantDetail Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaIdentityConditionalAccessPolicy. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaSubscribedSku. Object reference not set to an instance of an object.
WARNING: Omitting calls to Get-PrivilegedRole and Get-PrivilegedUser.
WARNING: Error running Get-MgBetaSubscribedSku. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaUserCount. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaPolicyAuthorizationPolicy. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaDirectorySetting. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaPolicyAdminConsentRequestPolicy. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaPolicyAuthenticationMethodPolicy. Object reference not set to an instance of an object.
WARNING: Error running Get-MgBetaDomain. Object reference not set to an instance of an object.
Fatal Error involving the Report Creation.
            Ending ScubaGear execution. See the exception message for more details: Invalid JSON primitive: .
At C:\Users\MyPC\Desktop\Other\O365\ScubaGear-1.1.0\PowerShell\ScubaGear\Modules\Orchestrator.psm1:917 char:13
+             throw $InvokeReportErrorMessage
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Fatal Error inv...ON primitive: .:String) [], RuntimeException
    + FullyQualifiedErrorId : Fatal Error involving the Report Creation.
            Ending ScubaGear execution. See the exception message for more details: Invalid JSON primitive: .

**Comment by amandaw33** [comment]
When I try to run Scuba, I have a bunch of warnings and no output file gets created. 

After the warnings, I see 2 errors about error running Connect-IPSSession about parameter showbanner and later a fatal error about Invalid JSON primitive in the Orchestrator module.

The warnings before my errors mostly mention command was not found in the module microsoft.graph.beta.identity.directorymanagement. I do have that module installed but when I go to Import-Module I get an error about could not load file azure.core...

Happy to try to troubleshoot this as I'd like to be able to run it. I did run the old version ~2 years ago without issue.

**Comment by ramarrgit** [comment]
Hi Addam Schroll,

As advised downgraded the Microsoft.Graph.* modules from 2.13.0 to 2.12.0.
AAD report ran fine.
Many thanks for the fix.
Leaving it to you close this issue as and when you see it fit. I am ok to close it.

Regards,
Rama

**Comment by gdasher** [comment]
We've released version 1.1.1 which pins the ms graph version at 2.12.0 pending a fix from microsoft for the backwards compatibility issue.
